src/lib.rs

```diff
@@ -129,7 +132,14 @@ impl Config {
 /// - `filename` doesn't exist
 /// - for any reason `filename` could not be read (maybe no read permission)
 pub fn run(config: Config) -> Result<(), Box<Error>> {
-    let mut file = File::open(config.filename())?;
+    // `io::stdin()` returns a `Stdin` struct, which implements the `Read` trait.
+    // `File::open<P: AsRef<Path>>(path: P)` returns `Result<File>` which can be
+    // unwrapped into a `File` struct, which also implements the `Read` trait.
+    // We want `file` to be of type `io::Read`, but it cannot be passed by value.
+    let mut file = match config.filename().as_ref() {
```

Comment:
I believe this should be a function by itself.

It would make the code more readable and this comment could be improved to be part of the documentation. What do you think? :)

Reply:
Agreed. Which name would be proper for this function in your opinion?

Reply:
Thinking about the function's purpose:

It would return a handler to the `text source`, be it a `File` or `stdin`.

Maybe something like `get_input_source` or `get_text`.

I also think the `file` variable should have its name change to be in accordance with the new behaviour.